### ISSUE TITLE
[bug] generate method not called for conan create

### ISSUE THREAD

**Issue by lpapp** [conan-io/conan#12920]
### Environment details

* Operating System+version:
* Compiler+version:
* Conan version:
* Python version:

### Steps to reproduce

conan create -pr my_profile . 1.0.0@

on my recipe with a generate method in it, it does not eventually get called even tough it sets up important things, for example:

```
    def generate(self):                                                                                                   
        tc = AutotoolsToolchain(self)                                           
        tc.configure_args.append(f"configure_dir={self.source_folder}")         
        tc.make_args.append("-j1")                                              
        tc.generate()                                                           
        deps = AutotoolsDeps(self)                                              
        deps.generate()                                                         
        if self.settings.os == 'Windows':                                       
            self.conf["tools.microsoft.bash:subsystem"] = "cygwin"              
            self.conf["tools.microsoft.bash:path"] = os.path.join(self.source_folder, 'cygwin64', 'bin', 'bash.exe')
```

This is using conan 1.57.0

**Comment by memsharded** [comment]
Hi @lpapp 

Thanks for your report.
First thing, I am practically 100% sure the ``generate()`` method runs when you run ``conan create``. If you put there some prints, or raise exceptions, you should see it.

I guess your main issue is that the ``conf`` values are not being defined. This is expected, ``conf`` values are read-only, they cannot be defined from recipes, they are intended to be defined in profiles, global.conf or command line, but never in recipes. Some ``tool_requires`` can define some ``conf_info`` for its consumers, but ``self.conf``, is read-only in recipes.

**Comment by lpapp** [comment]
I am sorry, but your 100% assumption is incorrect. I did put a print there to verify before submitting a bug report and it is indeed not printed.

Regarding read-only, someone mentioned on Slack that it is possible to write the config as they used it in that way before. So, I am baffled by these contradictory statements from two angles.

And if this is true, this sounds like a blocker for us. We definitely need to define this in the recipe because this is recipe specific. So, if this is the case, this should also be fixed.

To summarise, this is not a question, but bug report. Effectively, a blocker one for us to adopt v2 style.

**Comment by memsharded** [comment]
> I am sorry, but your 100% assumption is incorrect. I did put a print there to verify before submitting a bug report and it is indeed not printed.

That is super surprising. We would need a full recipe to reproduce, because that is working fine here. Can you please provide such reproducible recipe?

**Comment by memsharded** [comment]
> Regarding read-only, someone mentioned on Slack that it is possible to write the config as they used it in that way before. So, I am baffled by these contradictory statements from two angles.

Slack is not official support, but community (a great community by the way), but they might not have all the right information, specially regarding the 2.0-beta.

> And if this is true, this sounds like a blocker for us. We definitely need to define this in the recipe because this is recipe specific. So, if this is the case, this should also be fixed.

Conf is also per-package if you want to. If it is harcoded in the recipe, then it is not a conf. If you have a question, use case or request regarding this, better file a new ticket to discuss, lets focus this ticket on the execution (or not), of the ``generate()`` method.

**Comment by lpapp** [comment]
I am not using 2.0-beta. As mentioned, I am using 1.57.0.

Setting the bash path in the recipe for autotools on Windows: https://github.com/conan-io/conan/issues/12924

**Comment by memsharded** [comment]
> Slack is not official support, but community (a great community by the way), but they might not have all the right information, specially regarding the 2.0-beta.

Sorry, the ``conf`` is a new feature, intended to supersed env-vars configuration in 2.0. It is already available in 1.X as "under development" feature to allow the migration, sorry for the confusion, that is what I had in mind. Yes, definitely available in 1.X, and the ``generate()`` method is validated to run always in ``conan create`` for 1.X

**Comment by memsharded** [comment]
Hi @lpapp 

Have you been able to come up with a recipe, a whole conanfile.py, a project or something that we can try to use to reproduce this report? The ``generate()`` method is certainly being called at ``conan create`` time, so there should be something else that inhibits it to run. Just to clarify, this method is massively used nowadays in many different places, for example +600 recipes already use it in ConanCenter: https://github.com/conan-io/conan-center-index.

No idea of what could it be, having such full recipe or repo to reproduce could certainly help identifying the problem. Thanks!

**Comment by memsharded** [comment]
Hi @lpapp 

Any further feedback here? Any hint that would allow us to reproduce the issue?

**Comment by memsharded** [comment]
Closing as staled.
